libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -1031,7 +1031,7 @@ method rpcHandler*(g: GossipSub,
 
     g.mcache.put(msgId, msg)
 
-    if g.verifySignature and not msg.verify(peer.peerId):
+    if not g.anonymize and g.verifySignature and not msg.verify(peer.peerId):
```

Comment:
there are a few more combinations to cover here:

* when the message has a signature, it must be validated - we should never propagate a message with an invalid signature - that includes discarding messages with empty bytes and other signatures / keys that can't be parsed
* when the message has a seqno, it must be valid as well (ie exactly 8 bytes long)
* if `verifySignature` is turned on, the message _must_ have a signature (and implicitly, it must be valid)
* if `anonymize` is turned on, we could either be strict (don't allow messages with signatures) or lax (allow messages with signatures) - for now, the latter is enough - we can add the strict checking at some other time.

Reply:
Alright makes sense.